web-editor.mdx

```diff
@@ -0,0 +1,293 @@
+image or use an existing image from the repo.
+
+<Frame>
+  <img src="/images/editor/images-light.png" className="block dark:hidden" />
+  <img src="/images/editor/images-dark.png" className="hidden dark:block" />
+</Frame>
+
+Uploading an image can be done through the modal:
+
+<Frame>
+  <img src="/images/editor/upload-light.png" className="block dark:hidden" />
+  <img src="/images/editor/upload-dark.png" className="hidden dark:block" />
+</Frame>
+
+And you can preview an existing image before you add it.
+
+<Frame>
+  <img src="/images/editor/image-preview-light.png" className="block dark:hidden" />
+  <img src="/images/editor/image-preview-dark.png" className="hidden dark:block" />
+</Frame>
+
+### Editing images
+
+In order to edit images, you just have to hover over the image to see the **"Delete"**
+and **"Edit"** buttons on the top right of the image.
+
+<Frame>
+  <img src="/images/editor/edit-image-light.png" className="block dark:hidden" />
+  <img src="/images/editor/edit-image-dark.png" className="hidden dark:block" />
+</Frame>
+
+Clicking the **"Edit"** button lets you edit the attributes of the image.
+
+<Frame>
+  <img src="/images/editor/image-form-light.png" className="block dark:hidden" />
+  <img src="/images/editor/image-form-dark.png" className="hidden dark:block" />
+</Frame>
+
+If you want to update the source of an image to be a new image that you haven't yet
+uploaded, you have to first delete the image you're changing, and then add a new one
+using the [instructions above](#adding-images).
+
+## Editor modes
+
+In order to offer the most flexibility, the editor has three modes:
+
+<Frame caption="View mode toggle">
+  <img src="/images/editor/mode-toggle-light.png" className="block dark:hidden" />
+  <img src="/images/editor/mode-toggle-dark.png" className="hidden dark:block" />
+</Frame>
+
+### Visual Editor
+
+The **"Visual Editor"** is the typical WYSIWYG mode you'd see in something like Notion.
+It offers you a view into your docs in a fully editable way that reflects what the final
+page would look like on your main docs site.
+
+### Source Editor
+
+The **"Source Editor"** offers you a way to edit your MDX files in code, the same way
+you'd do in your IDE. This offers less flexibility, in that our components aren't available
+for auto-complete, but it does have two unique advantages.
+
+1. It allows you to edit props of components (see our [limitations below](#current-limitations))
+which is currently not available in **"Visual Editor"** mode yet.
+
+2. It allows you to correct syntax errors which might've appeared in your repo that
+might not be compatible with **"Visual Editor"** mode until you've fixed them.
+
+### Diff View
+
+The **"Diff View"** is a way to view the changes made to a specific document before
+committing it to your repository.
+
+This will help you see changes you've made along with formatting changes made by
+the editor.
+
+## Current limitations
```

Comment:
do openapi files work? should we add as a limitation

Reply:
Ahh very true, thank you kind sir.